### ISSUE TITLE
Can not Withdraw rewards/ Wrong reward balance

### ISSUE THREAD

**Issue by IlyaSofronov** [Emurgo/yoroi-frontend#1832]
Hello @SebastienGllmt :)
I have an issue here..
Before start of epoch 233 (i.e. epoch 232) I withdrew my earned rewards and **deregistered staking key** at the same time. 
In the begining of epoch 233 I got new reward 192ADA but can not withdraw it because WITHDRAW button has gone. 
Later in current epoch 233 I've delegated again to the same pool and Withdraw button showed up but I get an error when I try to withdraw.
How can I get remaining rewards?

Yoroi shows this error
![image](https://user-images.githubusercontent.com/26296741/102479845-064c5200-4092-11eb-9eef-e90973d21a19.png)

Adalite shows this error
![image](https://user-images.githubusercontent.com/26296741/100804592-bf3e4a00-345f-11eb-9eea-a49d2596f74f.png)

**Updated: As I know now I wasn't supposed to get the reward (and I did not get) but Yoroi showed differently and confused me**
I had ADA for fees.

**Comment by IlyaSofronov** [comment]
In current epoch 233 I've tried to delegate again to the same pool and Withdraw button showed up but I got an error
![image](https://user-images.githubusercontent.com/26296741/100805159-c3b73280-3460-11eb-9d8b-e782a4356d2d.png)

**Comment by nicarq** [comment]
@IlyaSofronov why did you deregister the key? how are you getting rewards without a key?

**Comment by nicarq** [comment]
<img width="605" alt="Screen Shot 2020-12-01 at 9 45 26 PM" src="https://user-images.githubusercontent.com/1622112/100821555-9f3a6680-341e-11eb-8df3-18ae61a17b33.png">

**Comment by IlyaSofronov** [comment]
@nicarq I wanted to move all funds to another wallet. So I withdrew current rewards and deregistered staking key to get KeyDeposit back. this is a natural thing. I have to recieve 2 rewards after undelegation because this is how the snapshot model works. I participated in the network snapshots past epochs, the rewards were counted and should be paid in the coming epochs

**Comment by nicarq** [comment]
@IlyaSofronov yeah, we added the option to deregister a staking key, coz of that, but the other side of the coin is that you cannot receive rewards if you don't have a staking key registered.

Please double check that you actually receiving rewards

**Comment by IlyaSofronov** [comment]
@nicarq In current epoch 233 I again registered staking key and delegating to a pool. But I can not withdraw reward for epoch 233 

**Updated: I have no reward for epoch 233, just Yoroi confusing me**

**Comment by nicarq** [comment]
@IlyaSofronov ah you didn't mention that you had registered the staking key again, so I thought that could be the issue. Maybe there is a timing thing as well with the registration here. 

Can you try to resync the wallet?

<img width="1236" alt="Screen Shot 2020-12-01 at 11 43 35 PM" src="https://user-images.githubusercontent.com/1622112/100829382-390a0f80-342f-11eb-89f5-1d48ef380727.png">

**Comment by IlyaSofronov** [comment]
@nicarq Resync and removing wallet dont help still get an error
![image](https://user-images.githubusercontent.com/26296741/100829751-b451dc80-3494-11eb-9d8e-bbf515c0b0d0.png)

**Comment by IlyaSofronov** [comment]
@nicarq I restored this wallet in adawallet.io and it shows Rewards 0ADA, very strange. But Yoroi and Adalite show reward. Lets wait some epochs to see how it is going..

**Updated: adawallet.io (does not use db-sync) shows correct reward (0ADA), Yoroi and Adalite not (192ADA) because of db-sync bug**

**Comment by IlyaSofronov** [comment]
@nicarq @SebastienGllmt I suppose I got reward shown because of the db-sync bug. Reward is not in ledger. In reality my reward went to dcfund or treasury because my key was deregistered and I could not get any reward

**Comment by IlyaSofronov** [comment]
@nicarq @SebastienGllmt Since the start of new epoch 234 I've got 135ADA. My staking key was registered so I believe there is no problem. Previos epoch 233 I got 192 fake ADA caused of db sync bug as my key was deregistered. So my true reward balance is 135ADA now. But Yoroi shows me reward balance is 328ADA (135 true+192false) and I can not withdraw 135ADA. Yoroi throws Error received from server while sending transaction.

From cardanoscan **(true)**
![image](https://user-images.githubusercontent.com/26296741/101311003-f8712280-3882-11eb-84ee-1899534c4de6.png)

From Yoroi **(false)**
![image](https://user-images.githubusercontent.com/26296741/101311035-0f177980-3883-11eb-864a-1ff10c564c95.png)

**Comment by Teralodom** [comment]
@nicarq I can confirm this happens to me as well. The wrong amount is shown in Yoroi, explorers like Cardanoscan display the information correctly.

The error that comes when broadcasting the transaction is of the like:
```{"error":{"response":"Error trying to send transaction: Error: Request failed with status code 400 - ApplyTxError [LedgerFailure (DelegsFailure (WithdrawalsNotInRewardsDELEGS (fromList [(RewardAcnt {getRwdNetwork = Mainnet, getRwdCred = KeyHashObj (KeyHash \"XXX\")},Coin XXX)])))]"}}```

It happens because the transaction created is higher than the actual rewards available.

The endpoint that brings incorrect information about the rewards is:
`https://iohk-mainnet.yoroiwallet.com/api/account/state`

So basically, anyone that deregistered a key an then registered again in a subsequent epoch should be affected by this issue as Yoroi calculates and adds all rewards regardless if the key is deregistered and the rewards were never distributed.

As I am using a hardware wallet with a passphrase, there is no way for me to go around this. So I guess I have to wait and hope for a fix or Daedalus to add the passphrase feature.

**Comment by nicarq** [comment]
thanks @Teralodom this seems like a bug from cardano-db-sync there is a newer version that seems to fix this issue, so hopefully it solves the problem

**Comment by IlyaSofronov** [comment]
Now I see my reward balance correctly. I didn't try to withdraw. Anyway I close this issue. Thanks for resolving that

**Comment by behzadmesri** [comment]
I have this problem. what should I do ? please help me . I can not do Withdraw with the Yoroi Mobile app, no with the Chrome extension in Bravo .
